InternPool unreachable when coercing extern function (`// use getExternFunc() instead`)

### Zig Version

0.13.0

### Steps to Reproduce and Observed Behavior

```zig
const c = @cImport({
    @cInclude("pwd.h");
});

pub fn main() !void {
    callback(c.getpwuid);
}

fn callback(cb: fn (c.uid_t) callconv(.C) ?*c.passwd) void {
    _ = cb(0);
}
```


this code works in some previous version, but panic compiler in 0.13.0.

change ?*c.passwd to [*c]c.passwd can fix the problem.



### Expected Behavior

give some error tips rather than panic

Cannot reproduce on `0.14.0-dev.302+d404d8a36`:
```
a.zig:19:15: error: expected type 'fn (c_uint) ?*cimport.struct_passwd', found 'fn (c_uint) callconv(.C) [*c]cimport.struct_passwd'
    callback(c.getpwuid);
             ~^~~~~~~~~
a.zig:19:15: note: calling convention 'C' cannot cast into calling convention 'Unspecified'
a.zig:22:17: note: parameter type declared here
fn callback(cb: fn (c.uid_t) ?*c.passwd) void {
                ^~~~~~~~~~~~~~~~~~~~~~~
```

> Cannot reproduce on `0.14.0-dev.302+d404d8a36`:
> 
> ```
> a.zig:19:15: error: expected type 'fn (c_uint) ?*cimport.struct_passwd', found 'fn (c_uint) callconv(.C) [*c]cimport.struct_passwd'
>     callback(c.getpwuid);
>              ~^~~~~~~~~
> a.zig:19:15: note: calling convention 'C' cannot cast into calling convention 'Unspecified'
> a.zig:22:17: note: parameter type declared here
> fn callback(cb: fn (c.uid_t) ?*c.passwd) void {
>                 ^~~~~~~~~~~~~~~~~~~~~~~
> ```

oh, forget the callconv(.C)，should be like this:
fn callback(cb: fn (c.uid_t) callconv(.C) ?*c.passwd) void {
_ = cb(0);
}

--- a/src/InternPool.zig
+++ b/src/InternPool.zig
@@ -9433,12 +9433,11 @@ pub fn getCoerced(
     switch (ip.indexToKey(val)) {
         .undef => return ip.get(gpa, tid, .{ .undef = new_ty }),
         .extern_func => |extern_func| if (ip.isFunctionType(new_ty))
-            return ip.get(gpa, tid, .{ .extern_func = .{
+            return ip.getExternFunc(gpa, tid, .{
                 .ty = new_ty,
                 .decl = extern_func.decl,
                 .lib_name = extern_func.lib_name,
-            } }),
-
+            }),
         .func => unreachable,
 
         .int => |int| switch (ip.indexToKey(new_ty)) {

--- a/test/behavior/extern.zig
+++ b/test/behavior/extern.zig
@@ -45,3 +45,16 @@ test "function extern symbol matches extern decl" {
 export fn another_mystery_function() u32 {
     return 12345;
 }
+
+extern fn c_extern_function() [*c]u32;
+
+test "coerce extern function types" {
+    const S = struct {
+        export fn c_extern_function() [*c]u32 {
+            return null;
+        }
+    };
+    _ = S;
+
+    _ = @as(fn () callconv(.C) ?*u32, c_extern_function);
+}

